include/stl2/detail/iterator/ostream_iterator.hpp

```diff
@@ -26,63 +26,58 @@
 #include <stl2/detail/memory/addressof.hpp>
 
 STL2_OPEN_NAMESPACE {
-	namespace detail {
-		///////////////////////////////////////////////////////////////////////////
-		// ostream_cursor [Implementation detail]
-		//
-		template <class T, class charT, class traits>
-		requires
-			models::Same<T, void> ||
-			models::StreamInsertable<T, std::basic_ostream<charT, traits>>
-		class ostream_cursor {
+	///////////////////////////////////////////////////////////////////////////
+	// ostream_iterator [ostream.iterator]
+	// Extension: ostream_iterator<void> accepts any streamable type.
+	// Not to spec:
+	// * StreamInsertable constraint is implicit
+	//   See https://github.com/ericniebler/stl2/issues/246
+	//
+	template <class T = void, class charT = char, class traits = std::char_traits<charT>>
+	requires
+		models::Same<T, void> ||
+		models::StreamInsertable<T, std::basic_ostream<charT, traits>>
+	class ostream_iterator {
 		public:
-			using difference_type = std::ptrdiff_t;
-			using ostream_type = std::basic_ostream<charT, traits>;
-			struct mixin : protected ebo_box<ostream_cursor> {
-				using difference_type = ostream_cursor::difference_type;
-				using char_type = charT;
-				using traits_type = traits;
-				using ostream_type = ostream_cursor::ostream_type;
-				using ebo_box<ostream_cursor>::ebo_box;
-			};
+			typedef ptrdiff_t difference_type;
+			typedef charT char_type;
+			typedef traits traits_type;
+			typedef std::basic_ostream<charT, traits> ostream_type;
+
+			ostream_iterator() = default;
```

Comment:
I prefer `constexpr ostream_iterator() noexcept = default;` so the compiler will shout loudly if we screw up this type's subobjects in the future.